### ISSUE TITLE
Windows installer feature detection fails when user is not en-US

### ISSUE THREAD

**Issue by vincent-da2o** [lando/lando#3490]
When trying to upgrade to v3.8.xx I am getting the installer error: "Feature not enabled, VirtualMachinePlatform and  Microsoft-Windows-Subsystem-Linux feature must be enabled.

But they are both enabled on my PC and I am not getting any issue with prior versions (v3.6.4, v3.6.5)

**Comment by maximilian-walter** [comment]
I have the same problem. As a workaround, I had to change the language of Windows from german to english. Then the feature-detection worked properly.

I think the bug was introduced with this commit: https://github.com/lando/lando/pull/3484/commits/df6fa94d90d2942721a86d32d287f3d4e1269772

The output of the command `dism.exe /online /get-features /format:table` seems to get localized, so the search for the string "Enabled" did not work in my case, because Windows printed "Aktiviert" (german for "enabled").

**Comment by Scoottykun** [comment]
Erm, i'm struggling with this as well, on windows 10 in my case.
I tried to change my whole windows language from French to English, but i'm still having the "features not enable" error on lando 3.8.1 installation.
The `dism.exe /online /get-features /format:table` command keep showing the features and their state in French strangely, but the windows interface is displayed in english.
Any guidance would be greatly appreciated !

**Comment by j0Shi82** [comment]
Ran into the same issue and switching to English locale did resolve it. Thanks!

@Scoottykun here's a screenshot of my working language settings. Maybe it helps.

![landogit](https://user-images.githubusercontent.com/39054178/213140998-f9320866-ef25-4416-86c8-e51122589d48.PNG)

**Comment by Scoottykun** [comment]
Hi, thanks but sadly it doesn't seems to work in my case, i don't know why :(  I tried a few language configution combo with a reboot at each change but i still have the issue on lando 3.6.1 install.

![Screenshot 2023-01-18 113116](https://user-images.githubusercontent.com/7600027/213150655-c1d9ab07-cca5-4355-bf7c-0bd78ef417ff.png)

Could this issue with the right locale detection be fixed in 3.8.2 please ?

Thanks !

**Comment by apapalillo** [comment]
I can confirm changing the Windows language to en-US fixed the issue 🤦🏼‍♂️

**Comment by Scoottykun** [comment]
Damn, i really don't understand what's wrong in my case. 
I also have the same error with the fresh 3.10.0 official release.

Here a more Explicit capture that show that my Features are both enabled :
![lando](https://user-images.githubusercontent.com/7600027/214544219-a223548e-bf49-463a-b3a5-89d1bb2fd05e.png)

Is it possible to find logs or more detail on the encountered errors on installation ?

Thanks in advance for your help.

**Comment by Scoottykun** [comment]
For anyone having the same issue than me, i could find what's was wrong.
I was simply switching the language of my non-administrator windows user account that does not have any effet... but what i really needed to do was to do it on the windows **administrator** account (which is 100% logical since we must install Lando through administrator privileges...).
Once this was done i could install Lando without any further issue !

**Comment by seyfcom** [comment]
Hi ! I tried the language thing too, I gives me another error.

In french I had the message
`Feature not enabled, VirtualMachinePlatform and Microsoft-Windows-Subsystem-Linux feature must be enabled.`

and after changing to US english and restating the computer, I have this
`Error
Access violation at address 76BDFC8E in module 'ucrtbase.dll'. Read of address 03A172CC.`

I did not change any of the other settings. Both features are enable and used. WSL is WSL2, is that ok ?

**Comment by seyfcom** [comment]
Hi ! after trying a few things... It worked out... When I had the Language window open... 
So putting it all in US English seems to be a good start.
Thanks everyone !

**Comment by pirog** [comment]
We are looking into ways to handle this. Would be great to be able to keep the features detection but will ditch it if needed.

**Comment by pirog** [comment]
ahhh very nice, looks like there is an `/english` flag you can pass into `DISM.exe` to force usage. will roll a release to fix this.

**Comment by pirog** [comment]
latest release seemed to work for me so going to close this out. recommend people try it out and provide feedback. can reopen if it doesnt seem to resolve the language issue
https://github.com/lando/lando/releases/tag/v3.11.0

**Comment by 1name** [comment]
> latest release seemed to work for me so going to close this out. recommend people try it out and provide feedback. can reopen if it doesnt seem to resolve the language issue https://github.com/lando/lando/releases/tag/v3.11.0

I downloaded the new release 3.11 and got the same error. Maybe other developers have a result?

**Comment by possiBri** [comment]
I, too, received the same error after attempting an install of 3.11 — Windows 10, was always using English (though did have German enabled to switch between, but everything in language settings are set to english). I am the only user on my laptop and I have admin privileges.

**Comment by skepsys** [comment]
Got the same error trying to install 3.11.0 on Win 10 (""Feature not enabled, VirtualMachinePlatform and Microsoft-Windows-Subsystem-Linux feature must be enabled."). Language is set to en_US.

**Comment by onetrev** [comment]
Hey @pirog, dropping in my feedback... I have a clean Win 11 Build 22621.1702 installation. Trying to install Lando `lando-x64-v3.18.0.exe` produces for me the same _"Feature not enabled, VirtualMachinePlatform and Microsoft-Windows-Subsystem-Linux feature must be enabled"_ error. This happens for me even if I switch my language to English-US (from English-Australia). :(

Based on the above and that others above are reporting the continuation of the issue, maybe this should be re-opened?

**Comment by onetrev** [comment]
Also, could solving this issues simultaneously solve [#3498](https://github.com/lando/lando/issues/3498), as it too seems to be language related?

**Comment by pirog** [comment]
@onetrev if you run `dism.exe /online /get-features /format:table /english` from `cmd.exe` what do you get? You may need to run as an administrator.

**Comment by onetrev** [comment]
Hey @pirog. Thanks so much for the follow up. So it says...
```
VirtualMachinePlatform: Enabled
Microsoft-Windows-Subsystem-Linux: Disabled
```
Now at first this makes me look like a clown 🤡. But wait, I believe I now know the issue for both myself and for some of the other more recent people reporting this...

If you install WSL2 as per [latest recommendation](https://learn.microsoft.com/en-us/windows/wsl/install) from Microsoft (just using `wsl --install`), then WSL as an **Optional Feature** is NOT enabled. Instead, the Microsoft Store version of WSL2 is installed. The Store version being the one you are supposed to use now. And that's why above the feature shows as "Disabled".

So I wonder if the Lando installer needs to either find a way to check for the Store version of WSL2 in addition, or indeed just drop the requirement check?

Note, on my main dev machine I ended up just sticking with Lando being installed in my WSL2 distro. That still seems to be the better way to roll. Unless I'm missing something, if you install Lando on Windows then Lando can't access your files in your Linux distro which is the recommended / far more performant way to go. I wrote a [blog post](https://creativelogic.biz/blog/lando-on-wsl2) last year to help people setup Lando directly in WSL2, as I'm a big Lando fan. But I'm looking forward to the new _Hyperdrive_ that will make my post irrelevant and the whole process much easier. :)

**Comment by pirog** [comment]
Ahhh interesting and good research @onetrev. I would **not** have expected that to happen. Do you know if there is a different feature we can look for if people `wsl --install`?. Perhaps it is now sufficient to check for JUST `VirtualMachinePlatform`?

**Comment by onetrev** [comment]
Agreed, very surprising. So, because WSL is now a store app I don't think it's possible to check to see if it's installed. At least I couldn't find any documentation to that effect. 

The only thing I could find was if a user wants to check if they are running the store version, then running from Windows CLI: `wsl --version` will print out a bunch of version information. If you are on the old _Windows Feature_ version of WSL and run that command it will just output the WSL help info. I'm only a web dev, zero Windows app / installer experience, but I would guess this isn't something the installer can check for. :)

Long story short, I feel like just checking for `VirtualMachinePlatform` is probably the best option. If that's enabled, then it's a very good chance the user has one of the two WSL options installed. Most importantly, it then means someone with WSL properly installed is not blocked from installing Lando.

**Comment by RicLeP** [comment]
I've just done a clean install of Windows 11 today - had the same issue from running `wsl --install` with it not being listing in the Windows features.

As a test I tried enablling with add new features GUI, after a restart `Microsoft-Windows-Subsystem-Linux` is now shown as `enabled` by `dism.exe`. It was very quick when I enabled the feature so I don't think it installed anything - not sure how to check this - so I think it just picked up my existing install and changed a flag somewhere. FIngers crossed it broke nothing.

Lando has now installed.

**Comment by AaronFeledy** [comment]
Yeah, it looks like `Microsoft-Windows-Subsystem-Linux` is WSL1, which once upon a time, needed to be installed before you could upgrade to WSL2, so it would make sense that anyone running WSL2 would have it enabled. Now that you can install WSL2 on its own, `Microsoft-Windows-Subsystem-Linux` may not be enabled. It appears that WSL2 is part of VirtualMachinePlatform so it will be available, but perhaps not set up.

The Docker Desktop installation likely handles enabling and setting up WSL2. Does Lando need to check for it at all?

**Comment by pirog** [comment]
@AaronFeledy i dont love the current way we check but i think it provides the best overall experience. We are working to decouple the docker installation from the installers we offer. Once that happens we will have a better set of tools to assess whether lando is "ready" or not.

**Comment by Aldo-f** [comment]
Same issue; installed with `wsl --install`. 

```sh 
dism.exe /online /get-features /format:table /english
VirtualMachinePlatform                      | Enabled
Microsoft-Windows-Subsystem-Linux           | Disabled
``` 

Can't you check it with `wsl -l -v`?  (or `wsl -v`)

```
> wsl -l -v
  NAME      STATE           VERSION
* Ubuntu    Stopped         2

> wsl -v
WSL-versie: 1.2.5.0
Kernelversie: 5.15.90.1
WSLg-versie: 1.0.51
MSRDC-versie: 1.2.3770
Direct3D-versie: 1.608.2-61064218
DXCore-versie: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows-versie: 10.0.22621.2215

```
